### PR TITLE
[5.2] [Guided Tours] Fix permission errors when recording user tour state

### PR DESCRIPTION
**Pull request by obuisard**
Pull Request for Issue #44080.

### Summary of Changes

The Guided Tours component, until now, did not have a dispatcher.
We do not want to have restrictions on the Ajax call when a tour is running so that any user having access to the administrator console can have its tour state recorder.

### Testing Instructions

Create a user with Manager access permissions.
Log as that user.
Either the welcome or the what's new tour should start.

### Actual result BEFORE applying this Pull Request

When dismissing the tour (in any way), you get the error:
image
The state of the tour is not recorded.

### Expected result AFTER applying this Pull Request

No more errors.
On an auto-starting tour, the state of the tour is recorded in #__user_profiles table.
If tours are logged, there should be entries in the action logs.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/language/en-GB/com_guidedtours.ini [hunk 1]
```diff
@@ -94,7 +94,9 @@ COM_GUIDEDTOURS_TYPE_REDIRECT_URL_DESC="Enter the relative URL of the page you w
 COM_GUIDEDTOURS_TYPE_REDIRECT_URL_LABEL="Relative URL"
 COM_GUIDEDTOURS_URL_LABEL="Relative URL"
 COM_GUIDEDTOURS_URL_DESC="Enter the relative URL of the page from where you want to Start the tour, e.g administrator/index.php?option=com_guidedtours&view=tours for the tours' list page."
+COM_GUIDEDTOURS_USERSTATE_BADTOURID="The Tour needs to have a positive id."
 COM_GUIDEDTOURS_USERSTATE_CONNECTEDONLY="Tour User state action is only for connected users."
 COM_GUIDEDTOURS_USERSTATE_STATENOTSAVED="Tour User state not saved for user %1$s tour %2$s."
 COM_GUIDEDTOURS_USERSTATE_STATESAVED="Tour User state saved for user %1$s tour %2$s."
+COM_GUIDEDTOURS_USERSTATE_WRONGCONTEXT="The Tour User State context is incorrect."
```
Comment (brianteeman) [inline administrator/language/en-GB/com_guidedtours.ini]:
```suggestion
COM_GUIDEDTOURS_USERSTATE_WRONGCONTEXT="The Tour User state context is incorrect."
```

consistency with other strings

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 9af535f71ea82e744d61863cfc1aa7194596c6f0<br /><br />Issue replicated and fix confirmed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44082">issues.joomla.org/tracker/joomla-cms/44082</a>.</sub>

**Comment by richard67** [comment]
I've restored @brianteeman 's previous test result in the issue tracker as the change after that was only a case change in a language string.

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on 8b919f5ee93087e2d7824f1e4d4af252c2362e85<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44082">issues.joomla.org/tracker/joomla-cms/44082</a>.</sub>

**Review by tecpromotion — APPROVED** [review]

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44082">issues.joomla.org/tracker/joomla-cms/44082</a>.</sub>

**Comment by Hackwar** [comment]
Thank you for your contribution @obuisard!

**Pull request merged**